upf/upf_pfcp_server.c

```diff
@@ -1228,11 +1232,29 @@ static uword
       /* Free messages that must be freed after the loop */
       pool_foreach (msg, psm->msg_pool)
```

Comment:
this whole loop is IMHO bogus. Instead of checking all messages, putting the indexes of the to be freed messages in a vector and the running only over that vector should be more efficient.

Reply:
That's true. But the point of this PR is also to verify whether we have timer wheel issues, and checking all of the timestamps is the easiest way there. In a followup, it makes sense to use FIFOs and check timestamps starting from the head instead of using timer wheels for tracking message expiration, so this loop will be removed.